src/main/webapp/resources/js/pages/user/components/UserDetailsPage.jsx

```diff
@@ -0,0 +1,196 @@
+import React, { useState } from "react";
+import { useParams } from "react-router-dom";
+import {
+  Button,
+  Form,
+  Input,
+  notification,
+  Select,
+  Space,
+  Switch,
+  Typography,
+} from "antd";
+import { formatDate } from "../../../utilities/date-utilities";
+
+import {
+  useGetUserDetailsQuery,
+  useEditUserDetailsMutation,
+} from "../../../apis/users/users";
+
+/**
+ * React component to display the user details page.
+ * @returns {*}
+ * @constructor
+ */
+export default function UserDetailsPage() {
+  const { userId } = useParams();
+  const { data, isSuccess } = useGetUserDetailsQuery(userId);
+  const [editUser] = useEditUserDetailsMutation();
+
+  const [formErrors, setFormErrors] = useState();
+
+  const onFormFinish = (values) => {
+    editUser({ userId: userId, ...values })
+      .unwrap()
+      .then((payload) => {
+        notification.success({
+          message: i18n("UserDetailsPage.notification.success"),
+        });
+      })
+      .catch((error) => {
+        notification.error({
+          message: i18n("UserDetailsPage.notification.error"),
+        });
+        setFormErrors(error.data);
+      });
+  };
+
+  if (isSuccess) {
```

Comment:
Don't use the if/else statements for the rendering portion. Instead within the return have `!isLoading ?` then the stuff to display if the loading is done otherwise display the `ContentLoading` component we have with a message such as `Loading user details`. So it would look like this:

`return ( 
{isLoading ?
<Space> .... </Space> :
<ContentLoading message="Loading user details" />}`

Reply:
Please see 5200c30b6c76bd8bf906ec362711b245693da389.